overlord/devicestate/handlers_serial.go

```diff
@@ -358,6 +355,23 @@ func prepareSerialRequest(t *state.Task, regCtx registrationContext, privKey ass
 			noNetworkRetryInterval := retryInterval / 2
 			return "", &state.Retry{After: noNetworkRetryInterval}
 		}
+		if httputil.CertExpiredOrNotValidYet(err) {
+			// If the cert is expired/not-valid yet that
+			// most likely means that the devices has no
+			// ntp-synced time yet. We will retry for up
+			// to 2048s (timesyncd.conf(5) says the
+			// maximum poll time is 2048s which is
+			// 34min8s).
+			if nTentatives > maxTentativesCertExpired {
+				return "", err
+			}
+			return "", &state.Retry{After: retryInterval}
```

Comment:
thank you, this is better, I wonder though if 1 minute is too slow initially? and a bit too fast after a while? pure exponential backoff is probably too aggressive slowing down for this use case: would this be a compromise?

5 tentatives at 30s
5 tentatives at 1m
5 tentatives at 2m
5 tentatives at 4m

this gives 37.5m?

or something like that, basically choosing the retry interval based on nTentatives?

Reply:
Thanks, updated for this now.